x-pack/packages/kbn-elastic-assistant/impl/knowledge_base/knowledge_base_settings_management/index.tsx

```diff
@@ -51,14 +57,24 @@ import { useCreateKnowledgeBaseEntry } from '../../assistant/api/knowledge_base/
 import { useUpdateKnowledgeBaseEntries } from '../../assistant/api/knowledge_base/entries/use_update_knowledge_base_entries';
 import { SETTINGS_UPDATED_TOAST_TITLE } from '../../assistant/settings/translations';
 import { KnowledgeBaseConfig } from '../../assistant/types';
+import {
+  isKnowledgeBaseSetup,
+  useKnowledgeBaseStatus,
+} from '../../assistant/api/knowledge_base/use_knowledge_base_status';
+
+interface Params {
+  dataViews: DataViewsContract;
+}
 
-export const KnowledgeBaseSettingsManagement: React.FC = React.memo(() => {
+export const KnowledgeBaseSettingsManagement: React.FC<Params> = React.memo(({ dataViews }) => {
   const {
     assistantFeatures: { assistantKnowledgeBaseByDefault: enableKnowledgeBaseByDefault },
     http,
     toasts,
   } = useAssistantContext();
   const [hasPendingChanges, setHasPendingChanges] = useState(false);
+  const { data: kbStatus, isFetched } = useKnowledgeBaseStatus({ http, resource: ESQL_RESOURCE });
+  const isKbSetup = isKnowledgeBaseSetup(kbStatus);
```

Comment:
@patrykkopycinski, this may need to be updated once https://github.com/elastic/kibana/pull/184885 is merged